src/indra_cogex/sources/cli.py

```diff
@@ -53,71 +71,129 @@ def _iter_resolvers() -> Iterable[Type[Processor]]:
+    ]
     config = {} if config is None else json.load(config)
-    paths = []
-    na = NodeAssembler()
+    edge_paths = []
+    node_assemblers = {}
     for processor_cls in _iter_resolvers():
         if not processor_cls.importable:
             continue
         click.secho(f"Checking {processor_cls.name}", fg="green", bold=True)
-        if not load_only:
-            if (
-                force
-                or not processor_cls.nodes_path.is_file()
-                or not processor_cls.nodes_indra_path.is_file()
-                or not processor_cls.edges_path.is_file()
-            ):
-                processor = processor_cls(**config.get(processor_cls.name, {}))
-                click.secho("Processing...", fg="green")
-                # FIXME: this is redundant, we get nodes twice
-                nodes = list(processor.get_nodes())
-                processor.dump()
+        # First, get all required paths, we'll need them in the next steps
+        processed = True
+        processor_import_paths = []
+        processor_to_assemble_paths = {}
+        for node_type in processor_cls.node_types:
+            (
+                proc_nodes_path,
+                nodes_indra_path,
+                _,
+            ) = processor_cls._get_node_paths(node_type)
+            if node_type in to_assemble:
+                # Store the INDRA nodes pickle path for assembly
+                processor_to_assemble_paths[node_type] = nodes_indra_path
             else:
-                click.secho(
-                    f"Loading cached nodes from {processor_cls.nodes_indra_path}",
-                    fg="green",
-                )
-                with open(processor_cls.nodes_indra_path, "rb") as fh:
-                    nodes = pickle.load(fh)
-            na.add_nodes(nodes)
-
-        paths.append((processor_cls.nodes_path, processor_cls.edges_path))
+                # These will be imported directly
+                nodes_paths_for_import.append(proc_nodes_path)
+                processor_import_paths.append(proc_nodes_path)
+            if not proc_nodes_path.exists() or not nodes_indra_path.exists():
+                processed = False
+        if not processor_cls.edges_path.exists():
+            processed = False
+        edge_paths.append(processor_cls.edges_path)
+        click.secho(
+            f"Identified node paths for assembly: {processor_to_assemble_paths}",
+            fg="blue",
+        )
+        click.secho(
+            f"Identified node paths for import: {processor_import_paths}",
+            fg="blue",
+        )
+        # Run the processor if needed
+        if force_process or (process and not processed):
+            try:
+                processor = processor_cls(**config.get(processor_cls.name, {}))
+            except FileNotFoundError as e:
+                if not skip_failed_processors:
+                    raise
+                click.secho(f"Failed: {e}", fg="red")
```

Comment:
More verbose: Failed: ..., skipping corresponding nodes and relations from further processing and import